main.go

```diff
@@ -0,0 +1,218 @@
+	onlyFirstResult := flags.Bool("o", false, "Retrieve only one application for this route")
+	err = flags.Parse(args[1:])
+	if err != nil {
+		return
+	}
+
+	if len(flags.Args()) == 0 || len(args) == 0 {
+		err = fmt.Errorf("please specify the required parameters")
+		return
+	}
+	hostName := flags.Args()[0]
+
+	hasApiEndpoint, err := cliConnection.HasAPIEndpoint()
+	if err != nil || !hasApiEndpoint {
+		err = fmt.Errorf("no API endpoint set")
+		return
+	}
+
+	loggedIn, err := cliConnection.IsLoggedIn()
+	if err != nil {
+		return
+	}
+	if !loggedIn {
+		err = fmt.Errorf("error: please log in to search for apps")
+		return
+	}
+
+	cfc, err := createCfClient()
+	if err != nil {
+		return
+	}
+
+	route, err := findRoute(cfc, hostName)
+	if err != nil {
+		return
+	}
+
+	err = lookup(cfc, route, *target, *onlyFirstResult, cliConnection)
+	if err != nil {
+		return
+	}
+}
+
+func (l lookupRoute) GetMetadata() plugin.PluginMetadata {
+	return plugin.PluginMetadata{
+		Name: "lookup-route",
+		Version: plugin.VersionType{
+			Major: 0,
+			Minor: 1,
+			Build: 0,
+		},
+		Commands: []plugin.Command{
+			{
+				Name:     "lookup-route",
+				HelpText: "Cloud Foundry CLI plugin to identify applications, a given route is pointing to.",
+				UsageDetails: plugin.Usage{
+					Usage: "cf lookup-route [-t] [-o] ROUTE_URL",
+					Options: map[string]string{
+						"t": "Target the org/space containing the route",
+						"o": "Query only the first result",
+					},
+				},
+			},
+		},
+	}
+}
+
+func createCfClient() (*client.Client, error) {
+	cfg, err := config.NewFromCFHome()
+	if err != nil {
+		return &client.Client{}, err
+	}
+
+	cfc, err := client.New(cfg)
+	if err != nil {
+		return &client.Client{}, err
+	}
+
+	return cfc, nil
+}
+
+func retrieveDomains(cfc *client.Client, domainName string) ([]*resource.Domain, error) {
+	domainOpts := client.NewDomainListOptions()
+	domainOpts.Names.Values = append(domainOpts.Names.Values, domainName)
+	domains, err := cfc.Domains.ListAll(context.Background(), domainOpts)
+	if err != nil {
+		return nil, err
+	}
+	return domains, nil
+}
+
+func parseDomain(cfc *client.Client, query string) (*resource.Domain, string, *url.URL, error) {
+	routeUrl, err := url.Parse(query)
+	if err != nil {
+		return &resource.Domain{}, "", &url.URL{}, err
+	}
+	if routeUrl.Scheme == "" {
+		return &resource.Domain{}, "", routeUrl, fmt.Errorf("please provide the url including the scheme")
+	}
+
+	domains, err := retrieveDomains(cfc, routeUrl.Hostname())
+	if err != nil {
+		return &resource.Domain{}, routeUrl.Hostname(), routeUrl, err
+	}
+
+	if len(domains) > 0 {
+		return domains[0], routeUrl.Hostname(), routeUrl, nil
+	}
+
+	urlParts := strings.SplitN(routeUrl.Hostname(), ".", 2)
+	if len(urlParts) < 2 {
+		return &resource.Domain{}, "", routeUrl, fmt.Errorf("'%s' is not a domain", routeUrl.Hostname())
+	}
+	hostName := urlParts[0] //Subdomain is not empty
+	domainName := urlParts[1]
```

Comment:
There us also [`strings.Cut`](https://pkg.go.dev/strings#Cut) that splits a string into two along a delimiter.

so you could write:
```suggestion
	hostName, domainName, found := strings.Cut(routeUrl.Hostname(), ".")
	if !found {
		return &resource.Domain{}, "", routeUrl, fmt.Errorf("'%s' is not a domain", routeUrl.Hostname())
	}
```

Reply:
Very good suggestion, thanks, have changed and tested and pushed.